### PR TITLE
curvebs/mds: add curvebs logicpool io metric

### PR DESCRIPTION
**Pull request by wuhongsong**
signed-off-by: hzwuhongsong [hzwuhongsong@corp.netease.com](mailto:hzwuhongsong@corp.netease.com)

### CODE REVIEW

**Comment by wuhongsong** [comment]
image

image

**Comment by wuhongsong** [comment]
cicheck

**Comment by wuhongsong** [comment]
cicheck

**Comment by wuhongsong** [comment]
cicheck

File: src/mds/topology/topology_metric.cpp [hunk 1]
```diff
@@ -209,7 +215,36 @@ void TopologyMetricService::UpdateTopologyMetrics() {
...
+                  << iterCsStat.readIOPS;
+        }
+        it->second->writeIOPS.set_value(writeIOPS);
+        it->second->writeRate.set_value(writeRate);
+        it->second->readRate.set_value(readRate);
+        it->second->readIOPS.set_value(readIOPS);
+
+        DVLOG(6) << "pool metrics, pid is: "
+            << pid << ", write iops: "
+            << writeIOPS << ", write bs: "
```
Comment (aspirer) [inline src/mds/topology/topology_metric.cpp:242]: bps?

File: src/mds/topology/topology_metric.cpp [hunk 2]
```diff
@@ -209,7 +215,36 @@ void TopologyMetricService::UpdateTopologyMetrics() {
...
+        for (auto iterCsStat : copysetStatsCluster) {
+            if (iterCsStat.logicalPoolId == pid) {
+                readRate += iterCsStat.readRate;
+                writeRate += iterCsStat.writeRate;
+                readIOPS += iterCsStat.readIOPS;
+                writeIOPS += iterCsStat.writeIOPS;
+            }
+            DVLOG(6) << "copyset Metrics, csid is: "
+                  << iterCsStat.copysetId << ", write iops: "
+                  << iterCsStat.writeIOPS << ", write bs: "
```
Comment (aspirer) [inline src/mds/topology/topology_metric.cpp:230]: bps?

Comment (aspirer) [reply src/mds/topology/topology_metric.cpp:230]: cluster readbps:

File: src/mds/topology/topology_metric.h [hunk 3]
```diff
@@ -176,6 +176,15 @@ struct LogicalPoolMetric {
     // bytes have alloced
     bvar::Status<uint64_t> logicalAlloc;
 
+    // bandwidth of reading
+    bvar::Status<uint32_t> readRate;
+    // bandwidth of writing
+    bvar::Status<uint32_t> writeRate;
```
Comment (aspirer) [inline src/mds/topology/topology_metric.h:182]: should use uint64_t

**Review by aspirer — CHANGES_REQUESTED** [review]

File: src/mds/topology/topology_metric.cpp (hunk 1 above)
Comment (wuhongsong) [reply src/mds/topology/topology_metric.cpp]: fix

File: src/mds/topology/topology_metric.cpp (hunk 2 above)
Comment (wuhongsong) [reply src/mds/topology/topology_metric.cpp]: fix

File: src/mds/topology/topology_metric.h (hunk 3 above)
Comment (wuhongsong) [reply src/mds/topology/topology_metric.h]: fix

**Comment by wuhongsong** [comment]
cicheck

**Review by SeanHai — APPROVED** [review]

**Review by aspirer — APPROVED** [review]

**Pull request merged**